source/material_model/rheology/frank_kamenetskii.cc

```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright (C) 2019 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/rheology/frank_kamenetskii.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      FrankKamenetskii<dim>::FrankKamenetskii ()
+      {}
+
+
+
+      template <int dim>
+      double
+      FrankKamenetskii<dim>::compute_viscosity (const double reference_viscosity,
+                                                const double temperature,
+                                                const unsigned int composition) const
+      {
+        // Frank-Kamenetskii viscosity approximation
+        //    viscosity = ref_v * A * B * exp(E * (0.5-(T/ref_T)/2)))
+        // ref_v: reference viscosity,
+        // A: prefactor, B: pre-limiter, E: adjusted viscosity ratio,
+        // ref_T: reference temperature, T: temperature.
+
```

Comment:
instead of a reference temperature that is read in from the input file please use this `const double reference_temperature = this->get_adiabatic_surface_temperature()`. This way the plugin uses the value of the global input parameter `set Adiabatic surface temperature = ...`, and we avoid having different reference temperatures for different plugins. We have made bad experiences in the past with different plugins defining different reference temperatures (and we are still trying to unify all plugins to use this new form). Also this form could be easily extended at some point in the future to allow for an adiabatic temperature profile (if we want to extend this to a depth-dependent reference temperature).